File: solutions/_225_implement_stack_using_queues/solution.py
from collections import deque


class MyStack:
    def __init__(self):
        self.q = deque()

    def push(self, x: int) -> None:
        self.q.append(x)

    def pop(self) -> int:
        return self.q.pop()

    def top(self) -> int:
        return self.q[len(self.q) - 1]

    def empty(self) -> bool:
        return len(self.q) == 0


# https://leetcode.com/problems/implement-stack-using-queues/

File: solutions/_225_implement_stack_using_queues/solution.java
package _225_implement_stack_using_queues;

import java.util.*;

class MyStack {
    Deque<Integer> q;

    public MyStack() {
        this.q = new ArrayDeque<>();
    }

    public void push(int x) {
        this.q.addLast(x);
    }

    public int pop() {
        return this.q.removeLast();
    }

    public int top() {
        return this.q.peekLast();
    }

    public boolean empty() {
        return this.q.size() == 0;
    }
}

/*
 * https://leetcode.com/problems/implement-stack-using-queues/
 */